tests/integration/buildcmd/build_integ_base.py

```diff
@@ -396,3 +396,145 @@ def _verify_invoke_built_functions(self, template_path, functions, error_message
                 self.assertIn(error_message, process_stderr)
             else:
                 self.assertEqual(0, process_execute.process.returncode)
+
+
+class CDKTestBase(BuildIntegBase):
+    @classmethod
+    def setUpClass(cls):
+        cls.cmd = cls.base_command()
+        integration_dir = Path(__file__).resolve().parents[1]
+        cls.test_data_path = Path(integration_dir, "testdata", "buildcmd", "CDK")
+
+    def setUp(self):
+        # To invoke a function created by the build command, we need the built artifacts to be in a
+        # location that is shared in Docker. Most temp directories are not shared. Therefore we are
+        # using a scratch space within the test folder that is .gitignored. Contents of this folder
+        # is also deleted after every test run
+        self.scratch_dir = str(Path(__file__).resolve().parent.joinpath(str(uuid.uuid4()).replace("-", "")[:10]))
+        shutil.rmtree(self.scratch_dir, ignore_errors=True)
+        os.mkdir(self.scratch_dir)
+
+        self.working_dir = tempfile.mkdtemp(dir=self.scratch_dir)
+        self.custom_build_dir = tempfile.mkdtemp(dir=self.scratch_dir)
+
+        self.default_build_dir = Path(self.working_dir, ".aws-sam", "build")
+
+    def tearDown(self):
+        self.custom_build_dir and shutil.rmtree(self.custom_build_dir, ignore_errors=True)
+        self.working_dir and shutil.rmtree(self.working_dir, ignore_errors=True)
+        self.scratch_dir and shutil.rmtree(self.scratch_dir, ignore_errors=True)
+
+        npm_projects = ["cdk-example-rest-api-gateway"]
+
+        for project in npm_projects:
+            project_path = self.get_project_path(project)
+            node_modules_path = os.path.join(project_path, "node_modules")
+            package_lock_path = os.path.join(project_path, "package-lock.json")
+            if os.path.exists(node_modules_path):
+                shutil.rmtree(node_modules_path, ignore_errors=True)
+            if os.path.exists(package_lock_path):
+                os.remove(package_lock_path)
+
+    def copy_source_to_temp(self, project_name):
+        source = self.get_project_path(project_name)
+        destination = str(Path(self.working_dir, "source"))
+        if not os.path.exists(destination):
+            self.temp_source = shutil.copytree(source, destination)
+
+    def get_command_list(
+        self,
+        build_dir=None,
+        base_dir=None,
+        manifest_path=None,
+        use_container=None,
+        parameter_overrides=None,
+        mode=None,
+        function_identifier=None,
+        debug=False,
+        cached=False,
+        cache_dir=None,
+        parallel=False,
+        container_env_var=None,
+        container_env_var_file=None,
+        build_image=None,
+        project_type=None,
```

Comment:
I'd suggest we also add `cdk_app` and `cdk_context` as optional parameters to inject into the command list